### PR TITLE
Fix schema with relationship to dump None

### PR DESCRIPTION
**Pull request by mahenzon**
Caused by #278

When `None` is passed to a schema, which contains relationships, it fails. Though on the previous versions it successfully returned data. That's because of direct use of `self._serialize`. The check for missing fields just doesn't happen.

If you try to run the new test, it will fail with this text:

```
=================================== FAILURES ===================================
_____ TestResponseFormatting.test_schema_with_relationship_processes_none ______

self = <tests.test_schema.TestResponseFormatting object at 0x7f5ede746278>

    def test_schema_with_relationship_processes_none(self):
>       data = unpack(CommentSchema().dump(None))

test_schema.py:103: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../venv/lib/python3.6/site-packages/marshmallow/schema.py:553: in dump
    result = self._serialize(processed_obj, many=many)
../venv/lib/python3.6/site-packages/marshmallow/schema.py:517: in _serialize
    value = field_obj.serialize(attr_name, obj, accessor=self.get_attribute)
../marshmallow_jsonapi/fields.py:249: in serialize
    return self._serialize(None, attr, obj)
../marshmallow_jsonapi/fields.py:256: in _serialize
    related_url = self.get_related_url(obj)
../marshmallow_jsonapi/fields.py:155: in get_related_url
    params = resolve_params(obj, self.related_url_kwargs, default=self.default)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

obj = None, params = {'id': '<id>'}, default = <marshmallow.missing>

    def resolve_params(obj, params, default=missing):
        """Given a dictionary of keyword arguments, return the same dictionary except with
        values enclosed in `< >` resolved to attributes on `obj`.
        """
        param_values = {}
        for name, attr_tpl in params.items():
            attr_name = tpl(str(attr_tpl))
            if attr_name:
                attribute_value = get_value(obj, attr_name, default=default)
                if attribute_value is not missing:
                    param_values[name] = attribute_value
                else:
                    raise AttributeError(
                        "{attr_name!r} is not a valid "
>                       "attribute of {obj!r}".format(attr_name=attr_name, obj=obj)
                    )
E                   AttributeError: 'id' is not a valid attribute of None

../marshmallow_jsonapi/utils.py:55: AttributeError
============================== 1 failed in 0.25s ===============================
```

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -50,3 +50,6 @@ _sandbox
 env
 venv
 .python-version
+
+# IDE
+.idea/
```
Comment (mahenzon) [inline .gitignore]: we need this folder in .gitignore file, because it's used in JetBrains IDEs, for example PyCharm

**Comment by mahenzon** [comment]
ouch, tests failed. do you really still support marshmallow 2? ok then, I'll make the test compatible..

**Review by iamareebjamal — APPROVED** [review]

**Comment by iamareebjamal** [comment]
Ping @sloria

File: .gitignore (hunk 1 above)
Comment (sloria) [reply .gitignore]: This belongs in your [global gitignore file](https://help.github.com/articles/ignoring-files/#create-a-global-gitignore).

**Review by sloria — COMMENTED** [review]
Sorry for the delay on this. Looks good. I'll go ahead and update this with the latest `dev`.

**Pull request merged**